cmd/kubermatic-installer/quickstart_kind_resources.go

```diff
@@ -0,0 +1,213 @@
+				NodePort: 31514,
+			},
+			corev1.ServicePort{
+				Name:        "https",
+				Protocol:    "TCP",
+				AppProtocol: ptr("https"),
+				Port:        443,
+				TargetPort: utilintstr.IntOrString{
+					Type:   1,
+					IntVal: 0,
+					StrVal: "https",
+				},
+				NodePort: 32394,
+			},
+		},
+		Selector: map[string]string{
+			"app.kubernetes.io/component": "controller",
+			"app.kubernetes.io/instance":  "nginx-ingress-controller",
+			"app.kubernetes.io/name":      "nginx",
+		},
+		Type:                  "NodePort",
+		ExternalTrafficPolicy: "Cluster",
+		IPFamilies: []corev1.IPFamily{
+			"IPv4",
+		},
+		IPFamilyPolicy:        ptr(corev1.IPFamilyPolicy("SingleStack")),
+		InternalTrafficPolicy: ptr(corev1.ServiceInternalTrafficPolicyType("Cluster")),
+	},
+}
+
+var kindNodeportProxyService = corev1.Service{
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      "nodeport-proxy-kind",
+		Namespace: "kubermatic",
+	},
+	Spec: corev1.ServiceSpec{
+		Ports: []corev1.ServicePort{
+			corev1.ServicePort{
+				Name:     "sni-listener",
+				Protocol: "TCP",
+				Port:     6443,
+				TargetPort: utilintstr.IntOrString{
+					IntVal: 6443,
+				},
+				NodePort: 30652,
+			},
+			corev1.ServicePort{
+				Name:     "tunneling-listener",
+				Protocol: "TCP",
+				Port:     8088,
+				TargetPort: utilintstr.IntOrString{
+					IntVal: 8088,
+				},
+				NodePort: 32121,
+			},
+		},
+		Selector: map[string]string{
+			"app.kubernetes.io/name": "nodeport-proxy-envoy",
+		},
+		Type:                  "NodePort",
+		ExternalTrafficPolicy: "Cluster",
+		IPFamilies: []corev1.IPFamily{
+			"IPv4",
+		},
+		IPFamilyPolicy:        ptr(corev1.IPFamilyPolicy("SingleStack")),
+		InternalTrafficPolicy: ptr(corev1.ServiceInternalTrafficPolicyType("Cluster")),
+	},
+}
+
+var kindLocalSeed = kubermaticv1.Seed{
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      "kubermatic",
+		Namespace: "kubermatic",
+	},
+	Spec: kubermaticv1.SeedSpec{
+		Country:  "DE",      // TODO: some clever heuristic or geolocation service?
+		Location: "Hamburg", // TODO: some clever heuristic or geolocation service?
+		Kubeconfig: corev1.ObjectReference{
+			Namespace: "kubermatic",
+			Name:      "kubeconfig-seed",
+		},
+		Datacenters: map[string]kubermaticv1.Datacenter{
+			"kubevirt": kubermaticv1.Datacenter{
+				Country:  "DE",      // TODO: some clever heuristic or geolocation service?
+				Location: "Hamburg", // TODO: some clever heuristic or geolocation service?
+				Spec: kubermaticv1.DatacenterSpec{
+					Kubevirt: &kubermaticv1.DatacenterSpecKubevirt{
+						DNSPolicy: "ClusterFirst",
+						Images: kubermaticv1.KubeVirtImageSources{
+							HTTP: &kubermaticv1.KubeVirtHTTPSource{
+								OperatingSystems: map[providerconfig.OperatingSystem]kubermaticv1.OSVersions{
+									providerconfig.OperatingSystemUbuntu: map[string]string{
+										// TODO: upload to some kubermatic hosted public service?
+										"22.04": "http://image-repo.kube-system.svc/images/ubuntu-22.04.img",
```

Comment:
the `image-repo` is not part of this proposal, I think it would be best to upload and serve some minimal VM from a public s3 bucket or whichever hosting we go with.

Reply:
Can be served from OCI registry, but first we have to add support for that in KubeVirt provider.

Reply:
I guess we can for now also serve it from `dev.kubermatic.io/quickstart/kubevirt-images`

Reply:
18 minutes is quite long. We will have to work on this. But maybe not in this PR.

Reply:
alternative, we can do this as well
https://github.com/kubermatic/kubermatic/blob/57a1d18f8ac2a8a968a49d74023eeea260383722/cmd/kubermatic-installer/local_kind_resources.go#L179-L182

Reply:
How much time would you save because of this?

Reply:
guestimate: we can save roughly 14 minutes?

measured by a sophisticated benchmark (=> me derping on my laptop, creating `image-repo` inside the `kind` cluster and uploading the VM image there manually) the time of single machine MD with 2CPU and 4GB RAM coming alive drops from ~18 minutes to ~4 minutes.

Downloading the VM image from `dev.kubermatic.io/quickstart/kubevirt-images` with my network with average throughput `5MB/s` is ~2 minutes, uploading to local `kind` cluster `image-repo` is a few seconds (I guess loopback for kind is that fast). However, this all can happen in parallel to installing KKP master stack, which takes also almost 8 minutes with all the readiness checks, so I would expect preparing `image-repo` for the `local command` to not add extra time.

Reply:
Is this thread a blocker to approving this PR?

Reply:
it contains a discussion for future performance improvements, important but not a blocker for this PR imho.